### ISSUE TITLE
authentication-database not honored any more in boot 3.1

### ISSUE THREAD

**Issue by goafabric** [spring-projects/spring-data-mongodb#4391]
Using Boot 3.1 that also updated to latest spring data, suddenly causes my application
to bail out with an "authentication error".

Digging in further in noticed that
"spring.data.mongdob.authentication-database" is not honored any more.

before  boot 3.1 we had

`
MongoPropertiesClientSettingsBuilderCustomizer

private void applyCredentials(MongoClientSettings.Builder builder) {
		if (this.properties.getUri() == null && this.properties.getUsername() != null
				&& this.properties.getPassword() != null) {
			String database = (this.properties.getAuthenticationDatabase() != null)
					? this.properties.getAuthenticationDatabase() : this.properties.getMongoClientDatabase();
			builder.credential((MongoCredential.createCredential(this.properties.getUsername(), database,
					this.properties.getPassword())));
		}
	}
`

since boot 3.1 we have

`
PropertiesMongoConnectionDetails

if (this.properties.getMongoClientDatabase() != null || this.properties.getReplicaSetName() != null
				|| this.properties.getAuthenticationDatabase() != null) {
			builder.append("/");
			if (this.properties.getMongoClientDatabase() != null) {
				builder.append(this.properties.getMongoClientDatabase());
			}
			else if (this.properties.getAuthenticationDatabase() != null) {
				builder.append(this.properties.getAuthenticationDatabase());
			}
`

Practically the "else if" never gets executed, because getMongoClientDatabase() above is never null.
I wonder if the "else if" just should be a simple "if".

Long story short =>
Is this an intended change .. i did not see document inside the release notes ? 
Or just a bug ?

The default for mongodb seems to be to use an "authentication database" of type admin by default
And at least with Docker, i did not find a way to change that
